synapse/logging/context.py

```diff
@@ -669,7 +669,7 @@ def g(*args, **kwargs):
     return g
 
 
-def run_in_background(f, *args, **kwargs):
+def run_in_background(f, *args, **kwargs) -> defer.Deferred:
```

Comment:
I think this isn't 100% true actually since if the result is *not* a `Deferred` than we return it raw, which seems confusing. I think we should likely wrap that in `defer.succeed`?

Reply:
Yeah, probably. I don't think we should lie about our return value here.

Reply:
I wrapped the raw return value in `defer.succeed`, I suspect it doesn't really make sense to call `run_in_background` with a sync function, but 🤷 This would also break when using `await`, so maybe we should make this an error instead?